source/slang/compiler.cpp

```diff
@@ -179,64 +179,92 @@ namespace Slang
         return Stage::Unknown;
     }
 
+    SlangResult checkExternalCompilerSupport(Session* session, PassThroughMode passThrough)
+    {
+        switch (passThrough)
+        {
+            case PassThroughMode::None:
+            {
+                // If no pass through -> that will always work!
+                return SLANG_OK;
+            }
+            case PassThroughMode::dxc:
+            {
+#if SLANG_ENABLE_DXIL_SUPPORT
+                // Must have dxc
+                return (session->getOrLoadSharedLibrary(SharedLibraryType::Dxc, nullptr) &&
+                    session->getOrLoadSharedLibrary(SharedLibraryType::Dxil, nullptr)) ? SLANG_OK : SLANG_E_NOT_FOUND;
```

Comment:
Very technically isn't `dxil.dll` not actually required? It would be nice if we could gate dxc support just on `dxcompiler.dll`, but then produce a loud warning if we are ever trying to invoke `dxcompiler.dll` when `dxil.dll` is not also available.

Reply:
https://blogs.msdn.microsoft.com/marcelolr/2018/03/06/using-the-github-dxcompiler-dll/

So it does appear if dxil is not strictly needed. Can look at adding a warning when dxil is not found.

Reply:
Not *strictly* needed, but the 99% case is that if a user wants DXIL and `dxil.dll` can't be loaded then they are going to be unhappy, so the warning should be loud (if we had an option to make certain warnings be errors by default with an option to disable them, this would be one...).

For anybody following along with this conversation at a later date, you might want to check out [this blog post](https://www.wihlidal.com/blog/pipeline/2018-09-16-dxil-signing-post-compile/) for an example of when/how a user might want to invoke dxc without having `dxi.dll` present, and what they'd need to do after the fact to make things work.